server/lib/pbench/test/unit/test_process_tb.py

```diff
@@ -16,183 +16,185 @@ def get(self, section: str, option: str):
                 return "Authorization-token"
-            assert False
+            assert (
+                False
+            ), f"Unexpected configuration option, {section!r}/{option!r}, in mocked config.get()"
 
         def mock_results_push(self, ctrl, tb, token):
             return
 
         expected_result = Results(nstatus=f": processed {tarball}\n", ntotal=1, ntbs=1)
         monkeypatch.setattr(os, "remove", mock_remove)
-        monkeypatch.setattr(Path, "glob", mock_glob)
         monkeypatch.setattr(MockConfig, "get", mock_get)
         monkeypatch.setattr(ProcessTb, "results_push", mock_results_push)
-        ptb = ProcessTb(MockConfig(), self.logger)
+        ptb = ProcessTb(MockConfig(), test_logger)
         res = ptb.process_tb()
 
         assert res == expected_result
 
-    def test_process_tb_zero(self, monkeypatch):
+    def test_process_tb_zero(self, monkeypatch, test_logger):
         """verify processing if there are no TBs without any failure"""
         tb_path = fixture_path / "tarball"
 
-        def mock_glob(*args, **kwargs):
+        def mock_glob(self, pattern):
             return []
 
         def mock_get(self, section: str, option: str):
             if option == "pbench-receive-dir-prefix":
                 return tb_path
             if option == "put-token":
                 return "Authorization-token"
-            assert False
+            assert (
+                False
+            ), f"Unexpected configuration option, {section!r}/{option!r}, in mocked config.get()"
+
+        def mock_results_push(self, ctrl, tb, token):
+            assert False, "Unexpected call to mocked result_push function"
 
         expected_result = Results(nstatus="", ntotal=0, ntbs=0)
         monkeypatch.setattr(Path, "glob", mock_glob)
         monkeypatch.setattr(MockConfig, "get", mock_get)
-        ptb = ProcessTb(MockConfig(), self.logger)
+        monkeypatch.setattr(ProcessTb, "results_push", mock_results_push)
+        ptb = ProcessTb(MockConfig(), test_logger)
         res = ptb.process_tb()
 
         assert res == expected_result
 
-    def test_multiple_process_tb(self, monkeypatch, caplog):
+    def test_multiple_process_tb(self, monkeypatch, caplog, test_logger):
         """verify tar balls processing at the time of Failure as well as success"""
-        tb_path = fixture_path
-
-        def mock_remove(*args):
-            pass
+        tb_path = fixture_path / "mix-tarball"
+        good_tb = tb_path / tarball
+        bad_tb = tb_path / bad_tarball
 
-        def mock_glob(*args, **kwargs):
-            tb_path1 = fixture_path / "tarball-002" / "log.tar.xz.md5"
-            tb_path2 = fixture_path / "bad_tarball-002" / "bad_log.tar.xz.md5"
-            return [tb_path1, tb_path2]
+        def mock_remove(path, *, dir_fd=None):
+            if path == good_tb or path == bad_tb:
+                pass
+            assert False, "Unexpected tar ball in mocked os.remove()"
```

Comment:
Again, you can do this all in a single assertion, and it's "more Pythonic" to ask if `path` is in a list of values:
```
            assert path in [good_tb, bad_tb], "Unexpected tar ball in mocked os.remove()"
```

Reply:
The simpler the assertion expression, the more important the message is.  Thus, when the expression is just `False`, the message is really important, because there's really no context.  However, in this case, we can get by without the message because, if the assertion fails, it will print the expression which is fairly expressive -- obviously, if the assertion failed, it was because the path didn't match either of the expected values, and hopefully PyTest will share what the value of `path` actually was.